### PR TITLE
mtls support

### LINKED ISSUE #695: URLSchemeCheck called twice both with initDone:true
```
fortio curl -c 1 -connection-reuse 1 -n 6 -loglevel verbose  -resolve-ip-type ip http://debug.fortio.org
```
2x back to back
```
17:43:19 V http_client.go:137> URLSchemeCheck &{TLSOptions:{Insecure:false CACert: Cert: Key: UnixDomainSocket:} URL:http://debug.fortio.org NumConnections:1 Compression:false DisableFastClient:false HTTP10:false DisableKeepAlive:false AllowHalfClose:false FollowRedirects:false initDone:true https:false Resolve: extraHeaders:map[] hostOverride: HTTPReqTimeOut:3s UserCredentials: ContentType: Payload:[] LogErrors:true ID:0 SequentialWarmup:false ConnReuseRange:[1 1] NoResolveEachConn:false Offset:0s Resolution:0.001}
```

(ditto with fortio load)

### LINKED ISSUE #719: option to require client cert (mtls)

### PR DESCRIPTION
**Pull request by ldemailly**
fixes #719 

- mtls option
- generate test client cert and ca and use in tests for mtls
- use log.LogRequest and log.TLSInfo updated there in 1.3.0 for showing peer cert in debug endpoint and log
- mtls for grpc too
- prep for 1.52
- fix #695 while at it : incompatible change: you can't reuse and HTTPOptions and change the scheme in the URL.
- fix `fortio server -h` continuing instead of showing flags.

also used/depended on https://github.com/fortio/log/pull/5/files / cleaned up duplicate LogRequest and TLSInfo

In addition to the unit tests:

```bash
% go run . server -mtls -cacert cert-tmp/clica.crt -cert cert-tmp/server.crt -key cert-tmp/server.key &
```
client not providing a cert:
```
% curl --cacert cert-tmp/ca.crt https://localhost:8080/debug
18:16:01 http: TLS handshake error from 127.0.0.1:63245: tls: client didn't provide a certificate
curl: (56) OpenSSL SSL_read: error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate, errno 0
```

vs - when providing client cert:

```
% curl --cacert cert-tmp/ca.crt --cert cert-tmp/client.crt --key cert-tmp/client.key https://localhost:8080/debug 
18:17:11 Debug: GET /debug HTTP/2.0 127.0.0.1:63318 ()  "curl/7.86.0" https TLS_AES_128_GCM_SHA256 "CN=fake-client"
Φορτίο version dev  go1.19.6 arm64 darwin echo debug server up for 1m52.6s on MacBook-Air.local - request from 127.0.0.1:63318 https TLS_AES_128_GCM_SHA256 "CN=fake-client"

GET /debug HTTP/2.0

headers:

Host: localhost:8080
Accept: */*
User-Agent: curl/7.86.0

body:

```
(note the CommonName in the output)

```

### CODE REVIEW

File: fgrpc/pingsrv.go [hunk 1]
```diff
@@ -74,12 +74,14 @@ func PingServer(port, cert, key, healthServiceName string, maxConcurrentStreams
...
-	if cert != "" && key != "" {
-		creds, err := credentials.NewServerTLSFromFile(cert, key)
+	if tlsOptions.Cert != "" && tlsOptions.Key != "" {
+		tlsCfg, err := tlsOptions.TLSConfig()
 		if err != nil {
 			log.Fatalf("Invalid TLS credentials: %v", err)
 		}
-		log.Infof("Using server certificate %v and key %v to construct TLS credentials", cert, key)
+		creds := credentials.NewTLS(tlsCfg)
+		log.Infof("Using server certificate %v and key %v to construct TLS credentials - mtls %v",
```
Comment (wiardvanrij) [inline fgrpc/pingsrv.go]: Not sure if you want to log this?

File: fhttp/http_test.go [hunk 2]
```diff
@@ -733,13 +733,12 @@ func TestBadURLStdClient(t *testing.T) {
...
 func TestDefaultPort(t *testing.T) {
-	// TODO: change back to fortio demo server once setup
-	url := "http://istio.io/" // shall imply port 80
+	url := "http://demo.fortio.org/" // shall imply port 80
 	opts := NewHTTPOptions(url)
 	cli, _ := NewFastClient(opts)
 	ctx := context.Background()
 	code, _, _ := cli.Fetch(ctx)
-	if code != 301 {
+	if code != 303 {
```
Comment (wiardvanrij) [inline fhttp/http_test.go]: Update next comment about expecting 301

File: fgrpc/pingsrv.go (hunk 1 above)
Comment (ldemailly) [reply fgrpc/pingsrv.go]: why not? it's just the file names like before + whether mtls is on or not

File: fhttp/http_test.go (hunk 2 above)
Comment (ldemailly) [reply fhttp/http_test.go]: opsa yes, I switched back to demo.fortio.org and cloudflare does 303 not 301 for https redirect

Comment (ldemailly) [reply fhttp/http_test.go]: fixed now, thx

File: fgrpc/pingsrv.go (hunk 1 above)
Comment (ldemailly) [reply fgrpc/pingsrv.go]: I can add "from" for clarity

Comment (ldemailly) [reply fgrpc/pingsrv.go]: @wiardvanrij see https://github.com/fortio/fortio/pull/720/commits/38dd76a85b9a2013c5ba96773fccb50c171cae6a - better?

Comment (wiardvanrij) [reply fgrpc/pingsrv.go]: Eh maybe I said something too quick. I thought it was the actual key itself that got logged.. all good

**Review by wiardvanrij — APPROVED** [review]

File: fgrpc/pingsrv.go (hunk 1 above)
Comment (ldemailly) [reply fgrpc/pingsrv.go]: np, it was ambiguous wording, hopefully clearer/better now (just that the `mutal ` might be hard to spot)

**Pull request merged**